Cpp_hands_on_programming/lib/abstract/Ipc_method.cc

```diff
@@ -0,0 +1,60 @@
+#include "Ipc_method.h"
+#include <iostream>
+
+int Ipc_method::getFileSize(std::string filename) {
+    int size;
+    std::ifstream in(filename, std::ios::binary);
+	if (!in) {
+		perror("getFileSize");
+		exit(EXIT_FAILURE);
+	}
+	in.seekg(0, std::ios::end);
+	size = in.tellg();
+    in.close();
+    return size;
+}
+
+int Ipc_method::readFile(std::string filename, int offset, std::vector<char> &buffer, int size) {
+    std::ifstream in(filename, std::ios::binary);
+	if (!in.is_open()) {
+		perror("ifstream::open\n");
+		return -1;
+	}
+	// move file position according to offset
+	in.seekg(offset, std::ios::beg);
+	if (!in) {
+		perror("seekg\n");
+		return -1;
+	}
+
+	in.read(&buffer[0], size);
+	// check read status, in.gcount condition set to make last read possible
+	if (!in && in.gcount() == size) {
+		perror("ifstream::read fail\n");
+		return -1;
+	}
+	in.close();
+	return in.gcount();
+}
+
+int Ipc_method::writeFile(std::string filename, int total, std::vector<char> &buffer, int size) {
```

Comment:
It would be nicer if buffer is passed by 'const'. Passing just by reference kind of suggests that it can be changed by this function